src/tool/hpcrun/ompt/ompt-interface.c

```diff
@@ -699,6 +698,43 @@ ompt_idle_blame_shift_request()
   ompt_register_idle_metrics();
 }
 
+//*****************************************************************************
+// map opid operations
+//*****************************************************************************
+
+// Records the (target id —> cct_node *) mapping in a thread local variable
+void
+hpcrun_opid_map_record_target(int target_id,
+                              cct_node_t *node)
+{
+  assert(target_id != 0);
+  ompt_region_map_insert((uint64_t) target_id, node);
+}
+
+
+// Adds an (opid, cct_node_t *) entry to the concurrent map by finding
+// the cct_node_t * associated with target_id from the thread_local variable
+void
+hpcrun_opid_map_insert(int host_op_id,
+                       int target_id)
+{
+  cct_node_t *node = ompt_region_map_entry_callpath_get(ompt_region_map_lookup(target_id));
+  ompt_host_op_map_insert(host_op_id, node);
+}
+
+
+// Look up the cct node associated with host_op_id in a map.
+// The map is mutable concurrent data structure. the GPU thread all CPU threads share it
+// this function will provide the pointer to the CCT node that represents the 
+// context for the target region. For now, this cct_node_t * will be in the thread’s CCT. 
+// Eventually, this function might return a pointer to a separate CCT that will be used by a device
+cct_node_t *
+hpcrun_opid_map_lookup(int host_op_id)
+{
+  // TODO(keren): There's no remove implementation in both host_op_map and region_map
```

Comment:
We are supposed to remove entries from the map. However, in both region_map and host_op_map, the remove operations are not implemented yet. It seems that we have to write a splay tree remove.